olp-cpp-sdk-dataservice-read/src/repositories/PartitionsCacheRepository.cpp

```diff
@@ -55,6 +55,14 @@ std::string CreateKey(const std::string& hrn, const std::string& layer_id,
 std::string CreateKey(const std::string& hrn, const int64_t catalogVersion) {
   return hrn + "::" + std::to_string(catalogVersion) + "::layerVersions";
 }
+
+time_t ConvertTime(std::chrono::seconds time) {
+  if (time == std::chrono::seconds::max()) {
+    return std::numeric_limits<time_t>::max();
+  }
+
+  return time.count();
```

Comment:
```suggestion
  return (time == std::chrono::seconds::max()) ? std::numeric_limits<time_t>::max() 
                                                                               : time.count();
```

Consider adding some constexpr defines for readability.